src/markerFinder.js

```diff
@@ -0,0 +1,109 @@
+module.exports = class MarkerFinder {
+  constructor(board) {
+    this.board = board
+    this.dimensions = board.dimensions
+
+    this.isOpen = this.isOpen.bind(this)
+  }
+
+  isOpen(i) {
+    return this.board.isOpen(i)
+  }
+
+  hasWinningSetup(winningSetup, marker) {
+    const matchingPositions = winningSetup.filter((num) => {
+      return this.board.indicesOf(marker).includes(num)
+    })
+
+    return (matchingPositions.length === (this.dimensions.boardSize - 1)) &&
+            (winningSetup.some(this.isOpen))
+  }
+
+  isGameOver(idealSetup, marker) {
+    const alignedMarkers =  idealSetup.filter((num) => {
+      return this.board.indicesOf(marker).includes(num)
+    })
+
+    return alignedMarkers.length === this.dimensions.boardSize
+  }
+
+  parseDownwardDiagonal(marker, onGameOver, onWinningSetup) {
+    const { downwardDiagonals, boardSize } = this.dimensions
+
+    if (this.isGameOver(downwardDiagonals, marker)) {
+      onGameOver({ marker, positions: downwardDiagonals })
```

Comment:
these callback functions being passed are the factory methods that create a new outcome. Although I think this might be a step in the right direction, I still think this class knows too much with `isGameOver` and `hasWinningSetup`. However if I extract these two functions into a `rules.js` file, I'm creating another module that is dependent on knowing the board. Any suggestions?